src/main/kotlin/no/nav/familie/ef/sak/avstemming/KonsistensavstemmingScheduler.kt

```diff
@@ -0,0 +1,54 @@
+package no.nav.familie.ef.sak.avstemming
+
+import no.nav.familie.ef.sak.repository.InsertUpdateRepository
+import no.nav.familie.ef.sak.repository.RepositoryInterface
+import no.nav.familie.ef.sak.repository.domain.Stønadstype
+import no.nav.familie.ef.sak.task.KonsistensavstemmingPayload
+import no.nav.familie.ef.sak.task.KonsistensavstemmingTask
+import no.nav.familie.prosessering.domene.TaskRepository
+import org.slf4j.LoggerFactory
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.Version
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Repository
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import kotlin.random.Random
+
+@Service
+class KonsistensavstemmingScheduler(private val repository: KonsistensavstemmingJobbRepository,
+                                    private val taskRepository: TaskRepository) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Scheduled(cron = "0 0 0/12 * * *")
+    @Transactional
+    fun opprettTasks() {
+        Thread.sleep(Random.nextLong(20)) // YOLO unngå feil med att 2 noder
+        val tidspunkt = LocalDate.now().plusDays(2)
+        val jobber = repository.findAllByOpprettetIsFalseAndTriggerdatoIsBefore(tidspunkt)
+        jobber.forEach {
+            val triggerTid = it.triggerdato.atTime(8, 0)
+            logger.info("Oppretter task for triggerTid=$triggerTid")
+            val payload = KonsistensavstemmingPayload(Stønadstype.OVERGANGSSTØNAD, triggerTid)
+            taskRepository.save(KonsistensavstemmingTask.opprettTask(payload))
```

Comment:
Denne vil feile og rulle tilbake hvis to jobber kjører samtidig..?

Reply:
Burde vi isåfall ha kjørt "update" på jobben først og catche optimisticLockException og logge dette til info? 🤔 

Selv med random er det en brukbar sjanse for at denne tryner?

Reply:
Ja, pga @Version så kommer `repository.updateAll(jobber.map { it.copy(opprettet = true) })` få optimistic lock exception. 

Jeg la til try catch i prosessering, men pga att man kan ha 2 tråder som kjører klart metoden samtidig, og då når de ska commite, så kommer 1 lyckes, og den andre feile. Selve "feilet" kastes utenfor denne metoden, inne i spring sin transaksjon. Vet ikke helt hvordan man skal løse det og tenker jeg skal prøve å få fikset det i prosessering (det schedule skjer ofte)

Ikke helt enkelt å teste dette heller 

Alternativ:
- lock på tabellen 
- har en metod i repository som endrer status på ett jobb (NY/LÅST/FERDIG), og som då oppdaterer fra ny til låst i en ny transaksjon, hvis det gikk fint, så oppretter man task og oppdaterer til FERDIG (ulempe hvis noe annet feiler og man då allerede har endret til LÅST)